tests/integ/conftest.py

```diff
@@ -63,26 +84,97 @@ def nonce(request):
     return request.config.getoption("--nonce")
 
 
+@pytest.fixture(scope="session")
+def dynamodb(do_provide_aws):
+    return fixed_session.client("dynamodb", endpoint_url=resource_options['endpoint_url'])
+
+
+@pytest.fixture(scope="session")
+def dynamodbstreams(do_provide_aws):
+    return fixed_session.client("dynamodbstreams", endpoint_url=resource_options['endpoint_url'])
+
+
+@pytest.fixture
+def engine(dynamodb, dynamodbstreams):
+    yield Engine(dynamodb=dynamodb, dynamodbstreams=dynamodbstreams)
+
+
+def get_open_port():
+    """
+    By using a socket and binding to "" (localhost), with a port of 0 (socket picks first non-privileged port
+    that is not in use, we can ensure that the port is available.
+    See:  https://stackoverflow.com/questions/2838244/get-open-tcp-port-in-python/2838309#2838309
+    and   https://stackoverflow.com/a/45690594
+    :return: The port to use
+    """
+    with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.listen(1)
+        port = s.getsockname()[1]
+        return port
+
+
+def get_ddb_local():
+    """
+    Ensure that dynamodb-local is downloaded and available.  It must match the sha256 in this file.
+    :return: The directory where it's installed.
+    """
+    localdir = pytest.config.rootdir.join('.dynamodb-local').strpath
+    if not os.path.exists(localdir):
```

Comment:
nit: print/logging statement would be nice before it starts downloading so we can debug without a stack trace, but travis is pretty good about providing context so it's not required.

Reply:
Added a message to the header that gets printed out.